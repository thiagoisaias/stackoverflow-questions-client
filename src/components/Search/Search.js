import React from "react";
import PropTypes from "prop-types";

import {
  InputGroup,
  InputListWrapper,
  Subtitle,
  StyledForm,
  StyledInput,
  StyledLabel,
  SubmitButton,
  SubmitButtonWrapper,
  Title,
  Wrapper
} from "./styled";

const Search = props => {
  const { handleInputChange, handleQuerySubmit, queryArgs } = props;

  const handleSubmit = event => {
    event.preventDefault();
    handleQuerySubmit();
  };

  return (
    <Wrapper>
      <Title>Stack Overflow</Title>
      <StyledForm onSubmit={handleSubmit}>
        <Subtitle>Buscar na API</Subtitle>
        <InputListWrapper>
          <InputGroup>
            <StyledLabel> Tags </StyledLabel>
            <StyledInput
              type="text"
              name="tags"
              onChange={handleInputChange}
              value={queryArgs.tags}
            />
          </InputGroup>
          <InputGroup>
            <StyledLabel> Limit </StyledLabel>
            <StyledInput
              type="number"
              name="limit"
              onChange={handleInputChange}
              value={queryArgs.limit}
            />
          </InputGroup>
          <InputGroup>
            <StyledLabel> Score </StyledLabel>
            <StyledInput
              type="number"
              name="score"
              onChange={handleInputChange}
              value={queryArgs.score}
            />
          </InputGroup>
          <InputGroup>
            <StyledLabel> Sort </StyledLabel>
            <StyledInput
              type="text"
              name="sort"
              // onChange={handleInputChange}
              // value={queryArgs.sort}
              value={"votes"}
              disabled
            />
          </InputGroup>
        </InputListWrapper>
        <SubmitButtonWrapper>
          <SubmitButton>Search</SubmitButton>
        </SubmitButtonWrapper>
      </StyledForm>
    </Wrapper>
  );
};

Search.propTypes = {
  handleInputChange: PropTypes.func.isRequired,
  handleQuerySubmit: PropTypes.func.isRequired,
  queryArgs: PropTypes.shape({
    limit: PropTypes.string.isRequired,
    score: PropTypes.string.isRequired,
    sort: PropTypes.string.isRequired,
    tags: PropTypes.string.isRequired
  })
};

export default Search;
