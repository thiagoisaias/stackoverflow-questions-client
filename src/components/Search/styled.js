import styled from "styled-components";
import { devices } from "../../utils/devices";

export const InputGroup = styled.div`
  display: flex;
  flex-direction: column;

  @media ${devices.mediumDown} {
    margin: 6px 0;
  }

  @media ${devices.largeUp} {
    margin: 0 6px;

    &:first-child {
      margin-left: 0;
    }

    &:last-child {
      margin-right: 0;
    }
  }
`;

export const InputListWrapper = styled.div`
  display: flex;
  justify-content: space-between;

  @media ${devices.mediumDown} {
    flex-direction: column;
  }
`;

export const StyledForm = styled.form`
  background-color: #ddd;
  padding: 16px;
`;

export const StyledInput = styled.input`
  display: block;
  height: 35px;
  width: calc(100% - 6px);
  margin-top: 4px;
  padding-left: 6px;
  font-size: 16px;
  border: none;
  color: #484848;
`;

export const StyledLabel = styled.label`
  color: #484848;
`;

export const SubmitButton = styled.button`
  height: 35px;
  width: 120px;
  outline: none;
  border: none;
  background-color: #8cc155;
  font-size: 16px;
  color: #fff;

  &:hover {
    cursor: pointer;
    background-color: #8cc155cc;
  }
`;

export const SubmitButtonWrapper = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-top: 24px;
`;

export const Subtitle = styled.p`
  color: #777;
  font-size: 18px;
  font-style: italic;
`;

export const Title = styled.h1`
  color: #484848;

  @media ${devices.mediumDown} {
    margin-left: 16px;
  }
`;

export const Wrapper = styled.div``;
