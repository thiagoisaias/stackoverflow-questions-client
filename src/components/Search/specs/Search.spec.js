import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Search from "../Search";

describe("Search", () => {
  const mockProps = {
    handleInputChange: jest.fn(),
    handleQuerySubmit: jest.fn(),
    queryArgs: {
      tags: "",
      limit: "",
      score: "",
      sort: ""
    }
  };
  const component = shallow(<Search {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
