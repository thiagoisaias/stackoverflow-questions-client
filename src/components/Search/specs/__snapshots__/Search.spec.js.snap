// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders properly 1`] = `
<styled.div>
  <styled.h1>
    Stack Overflow
  </styled.h1>
  <styled.form
    onSubmit={[Function]}
  >
    <styled.p>
      Buscar na API
    </styled.p>
    <styled.div>
      <styled.div>
        <styled.label>
           Tags 
        </styled.label>
        <styled.input
          name="tags"
          onChange={[Function]}
          type="text"
          value=""
        />
      </styled.div>
      <styled.div>
        <styled.label>
           Limit 
        </styled.label>
        <styled.input
          name="limit"
          onChange={[Function]}
          type="number"
          value=""
        />
      </styled.div>
      <styled.div>
        <styled.label>
           Score 
        </styled.label>
        <styled.input
          name="score"
          onChange={[Function]}
          type="number"
          value=""
        />
      </styled.div>
      <styled.div>
        <styled.label>
           Sort 
        </styled.label>
        <styled.input
          disabled={true}
          name="sort"
          type="text"
          value="votes"
        />
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.button>
        Search
      </styled.button>
    </styled.div>
  </styled.form>
</styled.div>
`;
